support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -23,6 +25,13 @@ type Props = {
 	abParticipations: Participations;
 };
 
+const ReturnAddressSchema = object({
+	link: string(),
+});
+export function setReturnAddress(link: InferInput<typeof ReturnAddressSchema>) {
```

Comment:
Why define `setReturnAddress` in `checkout.tsx` and export to `guardianAdLiteLanding.tsx`? It's not used in in `checkout.tsx`, so we could just define it in  `guardianAdLiteLanding.tsx` which would save us having to export it here and import it there?

Reply:
Yep makes sense to move out of `checkout.tsx` into `guardianAdLiteLanding.tsx`  Was in two minds over moving `ReturnAddressSchema` as its used in both `guardianAdLiteLanding.tsx` &  `checkout.tsx`.  Have moved it to keep it alongside the `setReturnAddress` (hopefully neater?)

Reply:
Yep I think the `ReturnAddressSchema` will only be needed where we set/read the return address session storage object. My suggestions made here https://github.com/guardian/support-frontend/pull/6741#discussion_r1933686924 and here https://github.com/guardian/support-frontend/pull/6741#discussion_r1933696607, would mean we set it on the Ad-Lite Landing Page and read it in the Ad-Lite TY Page Component that takes a user back to the return address.

I'm not sure how we define other session storage schemas? Are there other examples you've seen in our code?

We could possibly have a single module with these definitions, but I don't necessarily think we need to do that in this PR.

Reply:
Actually have moved most of sessionStorage code out into its own file now too, looks much neater. Think have fulfilled the suggestions too (see comments below).